packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,366 @@
+   *
+   * @default 'CloudWatchLogRule'
+   */
+  readonly schemaName?: string,
+
+  /**
+   * The value to sum up if the Rule is set to aggregate on SUM
+   *
+   * @default none
+   */
+  readonly sumValue?: string,
+
+  /**
+   * The version of the Schema. Should usually be 1
+   *
+   * @default 1
+   */
+  readonly version?: number,
+}
+
+/**
+ * Represents Contributor Insight Rule
+ */
+export interface IInsightRule extends IResource {
+
+  /**
+   * Name of this insight rule
+   *
+   * @attribute
+   */
+  readonly insightRuleRuleName: string;
+
+  /**
+   * Get a metric for this Insight Rule
+   *
+   * @param metricName The name of the metric to get
+   * @param metricOptions Additional metric options to apply
+   */
+  metric(metricName: string, metricOptions?: MathExpressionOptions): IMetric;
+}
+
+/**
+ * The base class for an Insight Rule
+ */
+abstract class InsightRuleBase extends Resource implements IInsightRule {
+  /**
+   * The name of this InsightRule
+   *
+   * @attribute
+   */
+  public abstract readonly insightRuleRuleName: string;
+
+  public metric(metricName: string, metricOptions: MathExpressionOptions = {}): IMetric {
+    return new MathExpression({
+      ...metricOptions,
+      expression: `INSIGHT_RULE_METRIC('${this.insightRuleRuleName}', '${metricName}')`,
+    });
+  }
+}
+
+/**
+ * A Contributor Insight Rule Construct
+ */
+export class InsightRule extends InsightRuleBase {
+
+  /**
+  * Import an existing Insight Rule from the rule name
+  *
+  * @param scope The parent creating construct (usually `this`)
+  * @param id The Construct's name
+  * @param insightRuleName The name of the Insight Rule
+  */
+  static fromInsightRuleName(scope: Construct, id: string, insightRuleName: string): IInsightRule {
+    class Import extends InsightRuleBase {
+      public readonly insightRuleRuleName = insightRuleName;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * Name of this Insight Rule
+   *
+   * @attribute
+   */
+  public readonly insightRuleRuleName: string;
+
+  /**
+   * ARN of this Insight Rule (i.e. arn:aws:cloudwatch:<region>:<account-id>:insight-rule/Foo)
+   *
+   * @attribute
+   */
+  public readonly insightRuleArn: string;
+
+  constructor(scope: Construct, id: string, props: InsightRuleProps) {
+    super(scope, id, {
+      physicalName: props.insightRuleName,
+    });
+
+    const ruleState = props.enabled === false ? 'DISABLED' : 'ENABLED';
+
+    const {
+      aggregateOn,
+      filters,
+      logGroupNames,
+      logFormat,
+      keys,
+      schemaName,
+      sumValue,
+      version,
+    } = props;
+
+    if (filters && filters.length > 4) {
+      throw new Error('Only 4 Filters can be applied to an Insight Rule');
+    }
+
+    if (aggregateOn === AggregateOptions.SUM && !sumValue) {
+      throw new Error('`sumValue` must be specified if using Aggregate Option SUM');
+    }
+
+    const ruleBody = JSON.stringify({
```

Comment:
@seanyles-amzn @HBobertz lets start here and discuss design of `ruleBody` because imo thats the most interesting part of this construct. I know we talked about not going down the RFC route but in that case, we should be able to communicate a coherent design in PR format. @seanyles-amzn what is your design for this part of the construct and what is your reasoning for going away from what was originally suggested in this comment: https://github.com/aws/aws-cdk/pull/19534/files#r834976361?

Reply:
i think that is a reasonable thing to do. however, what i'm looking for here is a bit of a paper trail that we can point to if someone asks about the design in the future. can you explain your design for RuleBody with code examples and show what the API should look like. we will then discuss and Hogan/I will give u the thumbs up. This will be helpful documentation especially because we do not have an RFC to back up our decisions.

Reply:
I will leave more details in the README in my next revision